docs/DeveloperGuide.md

```diff
@@ -1,34 +1,243 @@
+    1. Set up the correct JDK version for Gradle.
+        1. Click `Configure` > `Project Defaults` > `Project Structure`
+        1. Click `New...` and set it to the directory of the JDK.
+1. When prompted, **import** the project as a **Gradle project** (this can take a few minutes to complete). You can follow the following steps to import the project as a **Gradle Project**:
+
+    > :exclamation: **Note:** Importing a Gradle project is slightly different from importing a normal Java project.
+
+    1. IntelliJ IDEA by default has the Gradle plugin installed. If you have disabled it, click `File` > `Settings` > `Plugins` to enable them.
+    1. Click `Import Project` (or `Open or Import` in newer version of IntelliJ).
+    1. Locate the `build.gradle` file (not the root folder as you would do in normal importing) and select it. Click `OK`. If asked, choose to `Open as Project`, not `Open as File`.
+    1. Click `OK` to accept the default settings but do ensure that the selected version of `Gradle JVM` matches the JDK 11.
+
+### 2.3 Verifying Setup
+
+ 1. Run Fitr and enter a few commands to ensure that the application functions are as expected. You may refer to the _User Guide_ [here](https://ay2021s1-cs2113t-w13-2.github.io/tp/UserGuide.html) or enter `help` to find out what commands are supported in Fitr.
+ 1. Run the tests to ensure that they all pass, by executing the command `gradlew build` in IntelliJ's terminal.
+
+## 3. Design
+
+This section provides an overview of the design of the Fitr application.
+
+### 3.1 Architecture
+
+This section illustrates the high-level architecture of Fitr, as well as an illustration of the components used in Fitr.
+
+![Main class diagram](images/MainClassDiagram.png)
+
+_Figure 1: Architecture diagram of Fitr_
+
+Figure 1 above explains the high-level design of Fitr. Given below is a quick overview of each component.
+
+The main driver of the application is `Fitr` is responsible for mainly two phases:
+* At app launch: Initialises or loads the respective lists and user profile in the correct sequence and is in charge of connecting the various components with each other.
+* At shut down: Shuts down the components and invokes cleanup methods where necessary.
+
+The rest of the App consists of:
+* `Ui`: The user interface of the application.
+* `Parser`: Handles and parses user commands.
+* `Command`: Handles the various commands Fitr understands.
+* `User`: Contains and handles all information related to user data.
+* `StorageManager`: Reads data from and writes data back into a text file for future use.
+* `ListManager`: Handles all the list operations in Fitr.
+
+Figure 2 below shows how the components work with one another, when a user enters the following command `food chicken rice /600 1`:
+
+![Add food sequence diagram](images/AddFoodSequenceDiagram.png)
+
+_Figure 2: Sequence diagram when adding a food into Fitr_
+ 
+### 3.2 Components
```

Comment:
This part is written clearly and succinctly. Especially on the different behaviour when different inputs are given.